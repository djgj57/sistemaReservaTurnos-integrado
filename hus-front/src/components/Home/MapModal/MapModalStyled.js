import styled from "styled-components";
import { colors } from "./../../styles/utils/colors";
import { media } from "../../styles/utils/media";

export const Overlay = styled.div`
  width: 100%;
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 100;
`;

export const ModalContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  width: 538px;
  height: 402px;
  min-height: 100px;
  background: ${colors.primary};
  position: relative;
  border-radius: 5px;
  box-shadow: rgba(100, 100, 111, 0.5) 0px 7px 29px 0px;
  .titleContainer {
    display: none;
  }
  .mapRedirect {
    font-weight: 600;
    font-size: 1rem;
  }
`;
