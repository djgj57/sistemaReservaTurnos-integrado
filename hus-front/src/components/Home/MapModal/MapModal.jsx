import React from "react";
import { Overlay, ModalContainer } from "./MapModalStyled";
import ProductMap from "./../../Product/Map/ProductMap";

const MapModal = ({ mapModal, setMapModal, product, urlRedirect }) => {
  const defaultOptions = {
    loop: false,
    autoplay: true,
    animationData: require("./../../../assets/tick_success.json"),
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice",
    },
  };
  return (
    <>
      {mapModal && (
        <Overlay>
          <ModalContainer>
            <i
              class="fas fa-times"
              onClick={() => setMapModal(false)}
              style={{ position: "absolute", right: "20px", top: "20px" }}
            ></i>
            <ProductMap
              apiKey="AIzaSyAek8mB0pwqNVMFECvxjM3QK8gBILr0yJA"
              position={{ lat: product.latitude, lng: product.longitude }}
              city={product.city}
            />
            <a
              className="mapRedirect"
              target="_blank"
              rel="noreferrer"
              href={urlRedirect}
              style={{ fontWeight: "bold", cursor: "pointer" }}
            >
              IR A GOOGLE MAPS
            </a>
          </ModalContainer>
        </Overlay>
      )}
    </>
  );
};

export default MapModal;
