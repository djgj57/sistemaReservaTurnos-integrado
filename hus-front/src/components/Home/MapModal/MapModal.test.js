import React from "react";
import "@testing-library/jest-dom/extend-expect";
import {render} from "@testing-library/react";
import MapModal from "./MapModal";
import { MemoryRouter } from "react-router";

test ("SuccessfulBookingModal renders correctly", () => {
    const { container } = render (
        <MemoryRouter>
        <MapModal />
        </MemoryRouter>
    );
    expect (container).toMatchSnapshot ();
    });