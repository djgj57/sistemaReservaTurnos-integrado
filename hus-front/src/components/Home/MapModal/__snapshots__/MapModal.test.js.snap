// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessfulBookingModal renders correctly 1`] = `<div />`;
